doc/threshold/SONiC Threshold feature spec.md

```diff
@@ -0,0 +1,589 @@
+  
+## 3.2 DB Changes
+### 3.2.1 CONFIG DB
+
+#### THRESHOLD\_TABLE
+
+    ; New table
+    ; Defines threshold configuration.
+
+    key                      = buffer|type|alias|index ; buffer can be "priority-group" or "queue".
+                                                       ; type (buffer type) can be 
+                                                       ; "shared"
+                                                       ; "headroom"  
+                                                       ; "unicast"
+                                                       ; "multicast" 
+                                                       ; alias is unique across all DBs
+                                                       ; index (buffer index per port)
+    threshold                = 1*3DIGIT                ; Threshold in % (1-100) 
+
+    Example:
+   
+    Queue threshold configuration:
+    127.0.0.1:6379[4]> keys *THRESHOLD*
+    1) "THRESHOLD_TABLE|queue|unicast|Ethernet32|4"
+
+	127.0.0.1:6379[4]> HGETALL "THRESHOLD_TABLE|queue|unicast|Ethernet32|4"
+	1) "threshold"
+	2) "80"
+
+    Priority-group configuration:
+	127.0.0.1:6379[4]> keys *THRESHOLD*
+	1) "THRESHOLD_TABLE|priority-group|shared|Ethernet40|6"
+
+	127.0.0.1:6379[4]> hgetall THRESHOLD_TABLE|priority-group|shared|Ethernet40|6
+	1) "threshold"
+	2) "20"
+	
+
+### 3.2.2 APP DB
+N/A
+### 3.2.3 STATE DB
+N/A
+### 3.2.4 ASIC DB
+
+The ASIC DB is updated by SAI REDIS upon invocation of SAI REDIS APIs by ThresholdOrch.
+Following is an example of the ASIC\_DB configuration on applying a configuration of 80% threshold on queue 4 of port "Ethernet32"
+
+    127.0.0.1:6379[1]> keys *TAM*
+    1) "ASIC_STATE:SAI_OBJECT_TYPE_TAM_REPORT:oid:0x490000000005b1"
+    2) "ASIC_STATE:SAI_OBJECT_TYPE_TAM_TRANSPORT:oid:0x4c0000000005b3"
+    3) "ASIC_STATE:SAI_OBJECT_TYPE_TAM_COLLECTOR:oid:0x4e0000000005b4"
+    4) "ASIC_STATE:SAI_OBJECT_TYPE_TAM_EVENT:oid:0x5000000000063c"
+    5) "ASIC_STATE:SAI_OBJECT_TYPE_TAM_EVENT_THRESHOLD:oid:0x4a00000000063b"
+    6) "ASIC_STATE:SAI_OBJECT_TYPE_TAM_EVENT_ACTION:oid:0x4f0000000005b2"
+    7) "ASIC_STATE:SAI_OBJECT_TYPE_TAM:oid:0x3c00000000063d"
+
+    127.0.0.1:6379[1]> hgetall ASIC_STATE:SAI_OBJECT_TYPE_TAM_REPORT:oid:0x490000000005b1
+    1) "SAI_TAM_REPORT_ATTR_TYPE"
+    2) "SAI_TAM_REPORT_TYPE_PROTO"
+
+    127.0.0.1:6379[1]> hgetall ASIC_STATE:SAI_OBJECT_TYPE_TAM_TRANSPORT:oid:0x4c0000000005b3
+    1) "SAI_TAM_TRANSPORT_ATTR_TRANSPORT_TYPE"
+    2) "SAI_TAM_TRANSPORT_TYPE_UDP"
+    3) "SAI_TAM_TRANSPORT_ATTR_SRC_PORT"
+    4) "7070"
+    5) "SAI_TAM_TRANSPORT_ATTR_DST_PORT"
+    6) "7071"
+
+    127.0.0.1:6379[1]> hgetall ASIC_STATE:SAI_OBJECT_TYPE_TAM_COLLECTOR:oid:0x4e0000000005b4
+    1) "SAI_TAM_COLLECTOR_ATTR_SRC_IP"
+    2) "10.10.10.10"
+    3) "SAI_TAM_COLLECTOR_ATTR_DST_IP"
+    4) "127.0.0.1"
+    5) "SAI_TAM_COLLECTOR_ATTR_TRANSPORT"
+    6) "oid:0x4c0000000005b3"
+    7) "SAI_TAM_COLLECTOR_ATTR_DSCP_VALUE"
+    8) "0"
```

Comment:
where are this values get configured? do not see in the configuration db.

Reply:
Same response as earlier. Since this configuration is for receiving the threshold breach proto internally within SONIC, there is no configuration from user taken for the same.